### ISSUE TITLE
*/* doesn't ignore files for the first glob

### ISSUE THREAD

**Issue by huonw** [rust-lang/glob#29]
E.g.

```toml
# Cargo.toml
[package]

name = "test-glob"
version = "0.0.1"
authors = []

[dependencies]
glob = "*"
```

```rust
// src/main.rs
extern crate glob;
fn main() {
    for p in glob::glob("*/*").unwrap() {
        println!("{:?}", p);
    }
}
```

```
$ cargo run
   Compiling test-glob v0.0.1 (file:///home/huon/projects/test-rust/test-glob)
     Running `target/test-glob`
Ok(".git/HEAD")
Ok(".git/config")
Ok(".git/description")
Ok(".git/hooks")
Ok(".git/info")
Ok(".git/objects")
Ok(".git/refs")
Err(GlobError { path: ".gitignore", error: IoError { kind: OtherIoError, desc: "couldn\'t read directory", detail: Some("Not a directory; path=.gitignore") } })
Err(GlobError { path: "Cargo.lock", error: IoError { kind: OtherIoError, desc: "couldn\'t read directory", detail: Some("Not a directory; path=Cargo.lock") } })
Err(GlobError { path: "Cargo.toml", error: IoError { kind: OtherIoError, desc: "couldn\'t read directory", detail: Some("Not a directory; path=Cargo.toml") } })
Ok("src/main.rs")
Ok("target/.fingerprint")
Ok("target/build")
Ok("target/deps")
Ok("target/examples")
Ok("target/native")
Ok("target/test-glob")
```

The `Err`s should presumably just be ignored, since they don't satisfy the `*/*` glob (they're just files at the top level, and so only satisfy `*`).

**Comment by blaenk** [comment]
Thanks Huon. I'll take a look at this later tonight.

On Saturday, January 31, 2015, Huon Wilson <notifications@github.com> wrote:

> E.g.
>
> # Cargo.toml
> [package]
> name = "test-glob"version = "0.0.1"authors = []
>
> [dependencies]glob = "*"
>
> // src/main.rsextern crate glob;fn main() {
>     for p in glob::glob("*/*").unwrap() {
>         println!("{:?}", p);
>     }
> }
>
> $ cargo run
>    Compiling test-glob v0.0.1 (file:///home/huon/projects/test-rust/test-glob)
>      Running `target/test-glob`
> Ok(".git/HEAD")
> Ok(".git/config")
> Ok(".git/description")
> Ok(".git/hooks")
> Ok(".git/info")
> Ok(".git/objects")
> Ok(".git/refs")
> Err(GlobError { path: ".gitignore", error: IoError { kind: OtherIoError, desc: "couldn\'t read directory", detail: Some("Not a directory; path=.gitignore") } })
> Err(GlobError { path: "Cargo.lock", error: IoError { kind: OtherIoError, desc: "couldn\'t read directory", detail: Some("Not a directory; path=Cargo.lock") } })
> Err(GlobError { path: "Cargo.toml", error: IoError { kind: OtherIoError, desc: "couldn\'t read directory", detail: Some("Not a directory; path=Cargo.toml") } })
> Ok("src/main.rs")
> Ok("target/.fingerprint")
> Ok("target/build")
> Ok("target/deps")
> Ok("target/examples")
> Ok("target/native")
> Ok("target/test-glob")
>
> The Errs should presumably just be ignored, since they don't satisfy the
> */* glob (they're just files at the top level, and so only satisfy *).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/glob/issues/29>.
>


-- 
- Jorge Israel Peña

**Comment by huonw** [comment]
@blaenk thanks, but I went ahead and submitted a fix in #30.

**Comment by blaenk** [comment]
My hero!

On Saturday, January 31, 2015, Huon Wilson <notifications@github.com> wrote:

> @blaenk <https://github.com/blaenk> thanks, but I went ahead and
> submitted a fix in #30 <https://github.com/rust-lang/glob/pull/30>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/glob/issues/29#issuecomment-72346145>.
>


-- 
- Jorge Israel Peña
